src/ucp/wireup/address.c

```diff
@@ -284,9 +284,12 @@ static int ucp_address_pack_iface_attr(ucp_worker_h worker, void *ptr,
 
     packed->prio_cap_flags = ((uint8_t)iface_attr->priority);
     packed->overhead       = iface_attr->overhead;
-    packed->bandwidth      = iface_attr->bandwidth;
+    packed->bandwidth      = iface_attr->bandwidth.dedicated - iface_attr->bandwidth.shared;
     packed->lat_ovh        = iface_attr->latency.overhead;
 
+    /* check if at least one of bandwidth values is 0 */
+    ucs_assert((iface_attr->bandwidth.dedicated * iface_attr->bandwidth.shared) == 0);
```

Comment:
instead of assert pls check with 'if' and return error
comment that one of bw should be 0 because this is how we pack it

Reply:
added error evaluation